js/ViewPagerAndroid.ios.js

```diff
@@ -5,9 +5,119 @@
  * LICENSE file in the root directory of this source tree.
  *
  * @format
+ * @flow
  */
 
 'use strict';
 
-const UnimplementedView = require('react-native/Libraries/Components/UnimplementedViews/UnimplementedView');
-module.exports = UnimplementedView;
+const VIEWPAGER_REF = 'viewPager';
+
+import ReactNative, {UIManager} from 'react-native'
+import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';
+import React from "react";
+import NativeAndroidViewPager from './AndroidViewPagerNativeComponent';
+
+type TransitionStyle = 'scroll' | 'curl'
+type Orientation = 'horizontal' | 'vertical'
+
+type PageSelectedEvent = SyntheticEvent<
+  $ReadOnly<{|
+    position: number,
+  |}>,
+>;
+
+type Props = $ReadOnly<{|
+    /**
+     * Index of initial page that should be selected. Use `setPage` method to
+     * update the page, and `onPageSelected` to monitor page changes
+     */
+    onPageSelected?: ?(e: PageSelectedEvent) => void,
+    orientation?: ?Orientation,
+    transitionStyle?: ?TransitionStyle,
+    pageMargin?: ?number,
+    scrollEnabled?: ?number,
+    initialPage?: ?number,
+    style?: ?ViewStyleProp,
+  |}>;
+  
+  /**
+   * Container that allows to flip left and right between child views. Each
+   * child view of the `ViewPagerAndroid` will be treated as a separate page
+   * and will be stretched to fill the `ViewPagerAndroid`.
+   *
+   * It is important all children are `<View>`s and not composite components.
+   * You can set style properties like `padding` or `backgroundColor` for each
+   * child. It is also important that each child have a `key` prop.
+   *
+   * Example:
+   *
+   * ```
+   * render: function() {
+   *   return (
+   *     <ViewPagerAndroid
+   *       style={styles.viewPager}
+   *       initialPage={0}>
+   *       <View style={styles.pageStyle} key="1">
+   *         <Text>First page</Text>
+   *       </View>
+   *       <View style={styles.pageStyle} key="2">
+   *         <Text>Second page</Text>
+   *       </View>
+   *     </ViewPagerAndroid>
+   *   );
+   * }
+   *
+   * ...
+   *
+   * var styles = {
+   *   ...
+   *   viewPager: {
+   *     flex: 1
+   *   },
+   *   pageStyle: {
+   *     alignItems: 'center',
+   *     padding: 20,
+   *   }
+   * }
+   * ```
+   */
+
+function getViewManagerConfig(viewManagerName) {
+    if (!UIManager.getViewManagerConfig) {
+      // react-native <= 0.57
+      return UIManager[viewManagerName];
+    }
+    return UIManager.getViewManagerConfig(viewManagerName);
+  }
+
+  
+class ViewPagerAndroid extends React.Component<Props> {
+           
+    setPage = (selectedPage: number) => {
+      UIManager.dispatchViewManagerCommand(
+        ReactNative.findNodeHandle(this),
+        getViewManagerConfig('RNCViewPager').Commands.goToPage,
+        [selectedPage,true],
+      );
+    };
+  
+    setPageWithoutAnimation = (selectedPage: number) => {
```

Comment:
Is it API from Android implementation? If not i would rather stay with one method called `setPage` with 2 arguments `(selectedPage: number, animation: bool = true)`

Reply:
Now I want to match iOS API to Android API. I think, it will be unified in the future. 
in ViewManger there is a method 

```
RCT_EXPORT_METHOD(goToPage
                  : (nonnull NSNumber *)reactTag index
                  : (nonnull NSNumber *)index animated
                  : (BOOL)animated)
```
So I prepared this module for future api unifications.